Revision of the Deployment Recommendation document (#4295)
owncloud/docs: modules/admin_manual/pages/installation/deployment_recommendations.adoc

* command-line access or
* Cron access
* ideally both of the above

for three key reasons:

1. Without command-line access, xref:configuration/server/occ_command.adoc[OCC commands], required for administrative tasks such as repairs and upgrades, are not available.
2. Without Crontab access, you cannot run background jobs reliably.
xref:configuration/server/background_jobs_configuration.adoc#ajax[ajax/cron.php]
is available, but it is not reliable enough, because it only runs when people are using the web UI. Additionally, ownCloud relies heavily on xref:developer_manual:app/fundamentals/backgroundjobs.adoc[background jobs] especially for long-running operations, which will likely cause PHP timeouts.
3. Default PHP timeout values are often low. Having low timeout settings can break long-running operations, such as moving a huge folder.

== Scenario 1: Small Workgroups and Departments

This recommendation applies if you meet the following criteria:

[cols="30%,70%",options="header",]
|===
| Option
| Value

| Number of users
| Up to 150 users

| Storage size
| 100 GB to 10TB

| High availability level

> a| * Zero-downtime backups via Btrfs snapshots

Author: yes this is 🧙‍♂️ 😃 
it tells antora that the content of the table contains antora directives - which is in this case a list (*)

==== Operating system

Enterprise-grade Linux distribution with full support from an operating system vendor. We recommend Ubuntu 20.04, RedHat Enterprise Linux and SUSE Linux Enterprise Server 12+.

==== SSL Configuration

The SSL termination is done in Apache. A standard SSL certificate is required to be installed. See the {apache-ssl-howto-url}[official Apache documentation] or our xref:installation/letsencrypt/index.adoc[Let's Encrypt SSL Certificates] documentation.

==== Load Balancer

None.

==== Database

> MySQL, MariaDB, or PostgreSQL. We currently recommend MySQL / MariaDB, as our customers have had good experiences when moving to a Galera cluster to scale the DB. If using either MySQL or MariaDB, you must use the InnoDB storage engine as MyISAM is not supported, see: xref:configuration/database/linux_database_configuration.adoc#mysql-mariadb[MySQL / MariaDB storage engine]

Reviewer: ```suggestion
MySQL, MariaDB, or PostgreSQL. We currently recommend MySQL / MariaDB, as our customers have had good experiences when moving to a Galera cluster to scale the DB. If using either MySQL or MariaDB, you must use the InnoDB storage engine because MyISAM is not supported, see: xref:configuration/database/linux_database_configuration.adoc#mysql-mariadb[MySQL / MariaDB storage engine]
```

Reviewer: Just for quicker parsing of the sentence.

4.  Push database dump to backup.
5.  Start Apache.

After these steps have been completed, then, optionally, rsync the backup to either an external backup storage or tape backup. See xref:maintenance/index.adoc[the Maintenance section] of the Administration manual for tips on backups and restores.

==== Authentication

User authentication via one or several LDAP or Active Directory (AD) servers. See xref:admin_manual:configuration/user/user_auth_ldap.adoc[User Authentication with LDAP] for information on configuring ownCloud to use LDAP and AD.

==== Session Management

xref:configuration/server/caching_configuration.adoc#redis[Redis] is recommended and can be used for the session management storage.

Alternatively you can use local session management, see xref:local-session-management[Local Session Management].

==== Memory Caching

A memory cache speeds up server performance, and ownCloud supports a number of them. Refer to xref:admin_manual:configuration/server/caching_configuration.adoc[Configuring Memory Caching] for information on selecting and configuring a memory cache.

==== Storage

> Local storage, alternatively NFS if already available.

Reviewer: ```suggestion
Local storage or Network File System (NFS) if already available.
```

4.  Push database dump to backup.
5.  Start Apache.

After these steps have been completed, then, optionally, rsync the backup to either an external backup storage or tape backup. See xref:maintenance/index.adoc[the Maintenance section] of the Administration manual for tips on backups and restores.

==== Authentication

User authentication via one or several LDAP or Active Directory (AD) servers. See xref:admin_manual:configuration/user/user_auth_ldap.adoc[User Authentication with LDAP] for information on configuring ownCloud to use LDAP and AD.

==== Session Management

xref:configuration/server/caching_configuration.adoc#redis[Redis] is recommended and can be used for the session management storage.

Alternatively you can use local session management, see xref:local-session-management[Local Session Management].

==== Memory Caching

> A memory cache speeds up server performance, and ownCloud supports four of them. Refer to xref:admin_manual:configuration/server/caching_configuration.adoc[Configuring Memory Caching] for information on selecting and configuring a memory cache.

Reviewer 2: "four of them" - that count will go out-of-date as caching_configuration.adoc gets updated in future. Maybe:
```suggestion
A memory cache speeds up server performance, and ownCloud supports a number of them. Refer to xref:admin_manual:configuration/server/caching_configuration.adoc[Configuring Memory Caching] for information on selecting and configuring a memory cache.
```

==== Storage

Local storage, alternatively NFS if already available.

==== Recommmended Licensing Model

* Standard or Enterprise Edition
* See {owncloud-edition-url}[ownCloud Server or Enterprise Edition] for comparisons of the ownCloud editions.

== Scenario 2: Mid-Sized Enterprises

These recommendations apply if you meet the following criteria:

[cols="30%,70%",options="header",]
|===
| Option
| Value

| Number of users
| 150 to 1,000 users.

| Storage size
| Up to 200TB.

| High availability level

> a| * Every component is fully redundant and can fail
> * Backups without service interruption
> * Without service interruption.

Reviewer 2: ```suggestion
a| * Every component is fully redundant and can fail without service interruption
* Backups without service interruption
```

==== Authentication

User authentication via one or several LDAP or Active Directory servers. See xref:admin_manual:configuration/user/user_auth_ldap.adoc[User Authentication with LDAP] for information on configuring ownCloud to use LDAP and AD.

==== Session Management

xref:configuration/server/caching_configuration.adoc#redis[Redis] is recommended and can be used for the session management storage.

Alternatively you can use local session management, see xref:local-session-management[Local Session Management].

==== Memory Caching

> A memory cache speeds up server performance, and ownCloud supports four memory cache types. Refer to xref:admin_manual:configuration/server/caching_configuration.adoc[Configuring Memory Caching] for information on selecting and configuring a memory cache.

Reviewer 2: ```suggestion
A memory cache speeds up server performance, and ownCloud supports a number of memory cache types. Refer to xref:admin_manual:configuration/server/caching_configuration.adoc[Configuring Memory Caching] for information on selecting and configuring a memory cache.
```

==== Storage

For accessing a backend storage system via NFS, you can use a dedicated storage system like {netapp-hybrid-flash-array-url}[NetApp Hybrid Flash Storage Systems], or other systems like {ibm-elastic-storage-server-url}[IBM Elastic Storage] based on their Power8 servers or {redhat-ceph-url}[RedHat Ceph] with their NFS-Ceph gateway.

You may take a look on the {netapp-nfs-bpg-url}[NetApp NFS Best Practice and Implementation Guide] for best NFS configuring practices, especially section _9.4 Mount Option Best Practices with NFS_ on page 111 and {netapp-mysql-url}[MySQL Database on NetApp ONTAP] which also includes performance measurements.

==== Recommmended Licensing Model

* Enterprise Edition
* See {owncloud-edition-url}[ownCloud Server or Enterprise Edition] for comparisons of the ownCloud editions.

== Scenario 3: Large Enterprises and Service Providers

[cols="30%,70%",options="header",]
|===
| Option
| Value

| Number of users
| 5,000 to >100,000 users.

| Storage size
| Up to 1 petabyte.

| High availability level

> a| * Every component is fully redundant and can fail
> * Backups without service interruption
> * Without service interruption.|===

Reviewer 2: ```suggestion
a| * Every component is fully redundant and can fail without service interruption
* Backups without service interruption
```

==== Operating system

RHEL 7+ with latest service packs.

==== SSL Configuration

The SSL termination is done in the load balancer. A standard SSL certificate is needed, installed according to the load balancer documentation.

==== Load Balancer

> A redundant hardware load-balancer with heartbeat, for example, {bigip-f5-url}[F5 Big-IP]. This runs two load balancers in front of the application servers.

Reviewer: ```suggestion
A redundant hardware load balancer with heartbeat, for example, {bigip-f5-url}[F5 Big-IP]. This runs two load balancers in front of the application servers.
```

==== Authentication

> User authentication via one or several LDAP or Active Directory servers, or SAML/Shibboleth. See xref:configuration/user/user_auth_ldap.adoc[User Authentication with LDAP] and xref:enterprise/user_management/user_auth_shibboleth.adoc[Shibboleth Integration].

Reviewer: ```suggestion
User authentication via one or several LDAP or Active Directory servers or SAML/Shibboleth. See xref:configuration/user/user_auth_ldap.adoc[User Authentication with LDAP] and xref:enterprise/user_management/user_auth_shibboleth.adoc[Shibboleth Integration].
```

==== Storage

> For accessing a backend storage system via NFS, you can use a dedicated storage system like {netapp-hybrid-flash-array-url}[NetApp Hybrid Flash Storage Systems], or other systems like {ibm-elastic-storage-server-url}[IBM Elastic Storage] based on their Power8 servers or {redhat-ceph-url}[RedHat Ceph] with their NFS-Ceph gateway. Optionally, an S3 compatible object store can also be used.

Reviewer: ```suggestion
For accessing a backend storage system via NFS, you can use a dedicated storage system like {netapp-hybrid-flash-array-url}[NetApp Hybrid Flash Storage Systems] or other systems like {ibm-elastic-storage-server-url}[IBM Elastic Storage] based on their Power8 servers or {redhat-ceph-url}[RedHat Ceph] with their NFS-Ceph gateway. Optionally, an S3 compatible object store can also be used.
```

You may take a look on the {netapp-nfs-bpg-url}[NetApp NFS Best Practice and Implementation Guide] for best NFS configuring practices, especially section _9.4 Mount Option Best Practices with NFS_ on page 111 and {netapp-mysql-url}[MySQL Database on NetApp ONTAP] which also includes performance measurements.

==== Recommmended Licensing Model

* Enterprise Edition
* See {owncloud-edition-url}[ownCloud Server or Enterprise Edition] for comparisons of the ownCloud editions.

==== Redis Configuration

> Redis in a primary-replica configuration is {cloud-warm-hot-server-url}[a hot failover setup], and is usually sufficient. A replica can be omitted if high availability is provided via other means. And when it is, in the event of a failure, restarting Redis typically occurs quickly enough. Regarding Redis cluster, we don’t, usually, recommend it, as it requires a greater level of both maintenance and management in the case of failure.  A single Redis server, however, just needs to be rebooted, in the event of failure.

Reviewer: ```suggestion
Redis in a primary-replica configuration is {cloud-warm-hot-server-url}[a hot failover setup] and is usually sufficient. A replica can be omitted if high availability is provided via other means. If that's the case, restarting Redis typically happens fast enough in the event of a failure. Regarding Redis cluster, we don’t, usually, recommend it as it requires a greater level of both maintenance and management in the case of failure.  A single Redis server, however, just needs to be rebooted in the event of failure.
```
